### PR TITLE
refactor: update MSCB to use AutoExpand and setSelectedItemsOnTop

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

Related to https://github.com/vaadin/web-components/issues/6722#issuecomment-1803367987

- replaced `setGroupSelectedItems` with `setSelectedItemsOnTop`
- replaced `setAllChipsVisible` with `setAutoExpand` API

Note: this change needs the next web component alpha release. I created a PR to make reviewing easier.

## Type of change

- Refactor

### CODE REVIEW

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java [hunk 1]
```diff
@@ -395,25 +398,77 @@ public void updateSelection(Set<TItem> addedItems,
...
+     * @param {AutoExpandMode}
+     *            autoExpandMode
      * @since 24.3
-     * @param autoOpen
-     *            {@code true} to always show all chips
      */
-    public void setAllChipsVisible(boolean allChipsVisible) {
-        getElement().setProperty("allChipsVisible", allChipsVisible);
+    public void setAutoExpand(AutoExpandMode autoExpandMode) {
+        autoExpand = autoExpandMode;
```
Comment (knoobie) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java:451]: What is the expected behaviour if I pass `null`as mode? Does it reset the properties or fail with an exception? 

nit / styling: I would personally change the enum to take 2 parameter for `autoExpandHorizontally` and `autoExpandVertically` so the whole method could be reduced to getElement().set("a", autoExpandMode.getA()) + B.

Comment (web-padawan) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java:451]: > What is the expected behaviour if I pass nullas mode? Does it reset the properties or fail with an exception?

Good catch, thanks. I think it should fail.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java [hunk 2]
```diff
@@ -395,25 +398,79 @@ public void updateSelection(Set<TItem> addedItems,
...
+        NONE,
+    }
+
+    /**
+     * Gets the behavior of the component when not all selected items can be
+     * displayed as chips within the current field width.
      *
      * @since 24.3
-     * @return {@code true} if enabled, {@code false} otherwise
+     * @return {AutoExpandMode}
```
Comment (tomivirkki) [inline vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java:433]: The `@return` documentation doesn't look right:
Screenshot 2023-11-14 at 9 37 45

Could be something like `* @return The current {@link AutoExpandMode}`

Comment (web-padawan) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java]: Thanks, changed.

Comment (tomivirkki) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java]: It's still missing the `@link` from inside the brackets

Currently:
Screenshot 2023-11-14 at 10 13 13

Should be:
Screenshot 2023-11-14 at 10 16 52

Comment (web-padawan) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java]: Fixed.

File: vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java (hunk 1 above)
Comment (tomivirkki) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java:453]: > nit / styling: I would personally change the enum to take 2 parameter for autoExpandHorizontally and autoExpandVertically so the whole method could be reduced to getElement().set("a", autoExpandMode.getA()) + B.

I'd also recommend this refactor as it cleans up the setter quite a bit

Comment (web-padawan) [reply vaadin-combo-box-flow-parent/vaadin-combo-box-flow/src/main/java/com/vaadin/flow/component/combobox/MultiSelectComboBox.java:479]: > I'd also recommend this refactor as it cleans up the setter quite a bit

Thanks, done.

**Review by tomivirkki — APPROVED** [review]

**Review by DiegoCardoso — APPROVED** [review]

**Comment by web-padawan** [comment]
Merged to #5705 which contains the new web components alpha needed to make newly added ITs pass.

**Pull request merged**